src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/Native/SafeHandles.cs

```diff
@@ -114,34 +114,40 @@ public static void DeleteKeyContainer(SafeCertContextHandle pCertContext)
                 {
                     CRYPT_KEY_PROV_INFO* pProvInfo = (CRYPT_KEY_PROV_INFO*)pProvInfoAsBytes;
 
-                    // For UWP the key was opened in a CNG-only context, so it can/should be deleted via CngKey.Delete.
-                    // In all other contexts, the key was loaded into CAPI, so deleting it via CryptAcquireContext is
-                    // the correct action.
-#if NETNATIVE
-                    string providerName = Marshal.PtrToStringUni((IntPtr)(pProvInfo->pwszProvName));
-                    string keyContainerName = Marshal.PtrToStringUni((IntPtr)(pProvInfo->pwszContainerName));
-
-                    try
+                    if (pProvInfo->dwProvType == 0)
```

Comment:
I'd go with "stored in CNG" and "stored in CAPI", since handles aren't involved in cold storage, but otherwise good :)